packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +15,40 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	private caseInsensitiveSubCommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+		if (clientOptions.caseInsensitiveCommands) {
+			// Because slash commands must be lowercase anyways, blanketing all commands to lowercase is safe.
+			this.parsedSubcommandMappings.forEach((cmd) => cmd.name.toLowerCase());
+			this.caseInsensitiveSubCommands = true;
+		}
+
+		if (options.generateDashLessAliases) {
+			for (const mapping of this.parsedSubcommandMappings) {
+				if (!('messageRun' in mapping)) {
+					continue;
+				}
+
+				const dashLessAliases: string[] = [];
+
+				if (this.name.includes('-')) {
+					dashLessAliases.push(mapping.name.replace(/-/g, ''));
```

Comment:
Replace all?

Reply:
```suggestion
					dashLessAliases.push(mapping.name.replaceAll('-', ''));
```

Yeah Kyra recently taught me that apparently replaceAll comes in handy with some niche scenarios and is apparently under the hood not the same as the old code.